pandas/tests/test_frame.py

```diff
@@ -4815,6 +4815,24 @@ def test_repr_unicode(self):
         result = repr(df)
         self.assertEqual(result.split('\n')[0].rstrip(), ex_top)
 
+    def test_str_max_colwidth(self):
+        # GH 7856
```

Comment:
use ``with pd.option_context('max_colwidth',.....):`` so that it will only affect the code you want it to